include/istio/mixerclient/check_response.h

```diff
@@ -22,21 +22,15 @@
 namespace istio {
 namespace mixerclient {
 
-// The CheckResponseInfo holds response information in detail.
-struct CheckResponseInfo {
-  // Whether this check response is from cache.
-  bool is_check_cache_hit{false};
+// The CheckResponseInfo exposes policy and quota check details to the check
+// callbacks.
+class CheckResponseInfo {
+ public:
+  virtual ~CheckResponseInfo(){};
 
-  // Whether this quota response is from cache.
-  bool is_quota_cache_hit{false};
+  virtual const ::google::protobuf::util::Status& status() const = 0;
```

Comment:
question: do `status` and `routeDirective` need to be virtual?

Reply:
The check context shared pointer with all the state now implements the check response.   Before there was a copy into a result class which included a protobuf copy.   What's worse, a virtual call or a protobuf copy?   I figured the latter.

Reply:
Ok. I still think we could avoid it, but that probably requires more refactoring to erase mixerclient/envoy boundary than desired.

Reply:
I suppose I could create a wrapper object that delegates instead of making a virtual function call, but that would increment shared pointer ref counts.   Still, this feels like premature optimization to me.  For the gazillion inefficiencies of the mixer filter, replacing a virtual function call that I introduced to avoid some deep protobuf copies doesn't seem worth it